docs/src/pages/components/drawers/ResponsiveDrawer.js

```diff
@@ -166,7 +166,11 @@ export default function ResponsiveDrawer(props) {
 }
 
 ResponsiveDrawer.propTypes = {
-  // Injected by the documentation to work in an iframe.
-  // You won't need it on your project.
-  container: PropTypes.object,
+  /**
+   * Injected by the documentation to work in an iframe.
+   * You won't need it on your project.
+   */
+  container: PropTypes.object.isRequired,
```

Comment:
I don't think that it should be required, it warns in the docs and it will warn when people copy and paste the demo:
> Failed prop type: The prop `container` is marked as required in `ResponsiveDrawer`, but its value is `undefined`.

Reply:
Also, make me realize the tradeoff of these changes. As a maintainer, **when we want to edit the demos**, we have to browse the documentation with the TypeScript variant, because it's the source of truth that we edit. The prop types checks are no longer run. Checking that the TypeScript version works is no longer enough. We also need to check the JavaScript doesn't generate any warning.

So we trade a better UX for the users for a slower DX for the maintainers. It also creates an opportunity for small contributors to come and fix the prop-types warnings we introduce by mistake because we will miss some of them.

Reply:
> So we trade a better UX for the users for a slower DX for the maintainers. 

This will be solved once we no longer run the typescript version. The JS version should always stayed the source of truth. It was only changed because it was somehow not possible to run a task in parallel in a second terminal.

Since this also leads to bundle bloat and now negatively impacts this PR I'll revisit this and remove the redundant source. The typescript demo was never meant to be executed.

Reply:
Do you have an estimation of the bundle size overhead of bundling the TypeScript and JavaScript sources vs only the JavaScript one (not the raw, as I assume we will still need both to show in the source expand)?

Maybe the js to ts watch command could be run when launching `yarn docs:dev`?

Reply:
> If I understand it correctly, the TypeScript version will be the source of truth in the repository. 

That was never the intention. It was purely a documentation/type testing effort. It shouldn't have any impact beyond that.

> Do you have a % estimation of the bundle size overhead of bundling the TypeScript and JavaScript sources vs only the JavaScript one (not the raw, as I assume we will still need both to show in the source expand)?

between >0% and <=100%. 100% if only peer dependenncies are used (react + mui) and towards 0% the more dependencies are used.

And it has a perf/behavior impact since it remounts the demos when switching the version.

Reply:
Updated